engine/src/main/java/org/pentaho/di/trans/steps/webservices/wsdl/Wsdl.java

```diff
@@ -0,0 +1,401 @@
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.wsdl.Binding;
+import javax.wsdl.Definition;
+import javax.wsdl.Operation;
+import javax.wsdl.Port;
+import javax.wsdl.PortType;
+import javax.wsdl.Service;
+import javax.wsdl.WSDLException;
+import javax.wsdl.extensions.ExtensionRegistry;
+import javax.wsdl.factory.WSDLFactory;
+import javax.wsdl.xml.WSDLLocator;
+import javax.wsdl.xml.WSDLReader;
+import javax.xml.namespace.QName;
+
+import org.apache.http.auth.AuthenticationException;
+import org.pentaho.di.core.HTTPProtocol;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleStepException;
+import org.pentaho.di.core.logging.LogChannel;
+import org.pentaho.di.core.xml.XMLHandler;
+import org.w3c.dom.Document;
+
+/**
+ * Wsdl abstraction.
+ */
+public final class Wsdl implements java.io.Serializable {
+  private static final long serialVersionUID = 1L;
+  private Port _port;
+  private final Definition _wsdlDefinition;
+  private final Service _service;
+  private final WsdlTypes _wsdlTypes;
+  private HashMap<String, WsdlOperation> _operationCache;
+  private URI wsdlURI = null;
+
+  /**
+   * Loads and parses the specified WSDL file.
+   *
+   * @param wsdlURI
+   *          URI of a WSDL file.
+   * @param serviceQName
+   *          Name of the service in the WSDL, if null default to first service in WSDL.
+   * @param portName
+   *          The service port name, if null default to first port in service.
+   */
+  public Wsdl( URI wsdlURI, QName serviceQName, String portName ) throws AuthenticationException {
+    this( wsdlURI, serviceQName, portName, null, null );
+  }
+
+  public Wsdl( URI wsdlURI, QName serviceQName, String portName, String username, String password ) throws AuthenticationException {
+
+    this.wsdlURI = wsdlURI;
+    try {
+      _wsdlDefinition = parse( wsdlURI, username, password );
+    } catch ( AuthenticationException ae ) {
+      // throw this again since KettleException is catching it
+      throw ae;
+    } catch ( WSDLException e ) {
+      throw new RuntimeException( "Could not load WSDL file: " + e.getMessage(), e );
+    } catch ( KettleException e ) {
+      throw new RuntimeException( "Could not load WSDL file: " + e.getMessage(), e );
+    }
+    if ( serviceQName == null ) {
+      _service = (Service) _wsdlDefinition.getServices().values().iterator().next();
+    } else {
+      _service = _wsdlDefinition.getService( serviceQName );
+      if ( _service == null ) {
+        throw new IllegalArgumentException( "Service: "
+          + serviceQName + " is not defined in the WSDL file " + wsdlURI );
+      }
+    }
+
+    if ( portName == null ) {
+      _port = getSoapPort( _service.getPorts().values() );
+    } else {
+      _port = _service.getPort( portName );
+      if ( _port == null ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 108 to != check before setting the field value on line 112 or verify if the check is needed at all.